cmd/postcli/main.go

```diff
@@ -138,6 +129,24 @@ func processFlags() error {
 		return fmt.Errorf("invalid commitmentAtxId: %w", err)
 	}
 
+	if flagSet["labelsPerUnit"] && (cfg.LabelsPerUnit != config.MainnetConfig().LabelsPerUnit) {
+		fmt.Println("WARNING: labelsPerUnit is set to a non-default value. This makes the initialization incompatible " +
+			"with mainnet. If you're trying to initialize for mainnet, please remove the -labelsPerUnit flag")
+		if !yes {
+			fmt.Println("If you're sure you want to continue, please run with -yes")
+			os.Exit(1)
+		}
```

Comment:
Maybe it could prompt for `y/N` if `-yes` is not set, similarly to other tools like apt etc.?